src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/TempPE/DesignTimeInputsBuildManagerBridge.cs

```diff
@@ -0,0 +1,141 @@
+        /// </summary>
+        public string[] GetTempPEMonikers()
+        {
+            // Not using use the ThreadingService property because unit tests
+            _threadingService.VerifyOnUIThread();
+
+            Initialize();
+
+            return AppliedValue.Value.Inputs.ToArray();
+        }
+
+        /// <summary>
+        /// Gets the XML that describes a TempPE DLL, including building it if necessary
+        /// </summary>
+        public async Task<string> GetDesignTimeInputXmlAsync(string file)
+        {
+            // Not using use the ThreadingService property because unit tests
+            await _threadingService.SwitchToUIThread();
+
+            if (!SkipInitialization)
+            {
+                await InitializeAsync();
+            }
+
+            DesignTimeInputsDelta value = AppliedValue.Value;
+
+            return await _designTimeInputsCompiler.GetDesignTimeInputXmlAsync(file, value.TempPEOutputPath, value.SharedInputs);
+        }
+
+        /// <summary>
+        /// ApplyAsync is called on the UI thread and its job is to update AppliedValue to be correct based on the changes that have come through data flow after being processed
+        /// </summary>
+        protected override async Task ApplyAsync(IProjectVersionedValue<DesignTimeInputsDelta> value)
+        {
+            // Not using use the ThreadingService property because unit tests
+            await JoinableFactory.SwitchToMainThreadAsync();
+
+            DesignTimeInputsDelta delta = value.Value;
+
+            ImmutableHashSet<string>? removedFiles = AppliedValue?.Value.Inputs.Except(delta.Inputs);
+
+            // As it happens the DesignTimeInputsDelta contains all of the state we need
+            AppliedValue = value;
+
+            foreach (DesignTimeInputFileChange change in delta.ChangedInputs)
+            {
+                _buildManager.OnDesignTimeOutputDirty(change.File);
+            }
+
+            if (removedFiles != null)
+            {
+                foreach (string item in removedFiles)
+                {
+                    _buildManager.OnDesignTimeOutputDeleted(item);
+                }
+            }
+        }
+
+        /// <summary>
+        /// InitializeInnerCoreAsync is responsible for setting an initial AppliedValue. This value will be used by any UI thread calls that may happen
+        /// before the first data flow blocks have been processed. If this method doesn't set a value then the system will block until the first blocks
+        /// have been applied.
+        /// </summary>
+        protected override Task InitializeInnerCoreAsync(CancellationToken cancellationToken)
+        {
+            return Task.CompletedTask;
+        }
+
+        protected override Task DisposeCoreAsync(bool initialized)
+        {
+            if (initialized)
+            {
+                _designTimeInputsCompiler.Dispose();
```

Comment:
This is an import, are you deliberately disposing this? MEF components don't own the lifetime of their imports.